src/Control/Comonad/Cofree.purs

```diff
@@ -66,7 +68,16 @@ unfoldCofree
   -> s
   -> Cofree f a
 unfoldCofree e n s =
-  Cofree (e s) (defer \_ -> unfoldCofree e n <$> n s)
+  Cofree (defer \_ -> Tuple (e s) (unfoldCofree e n <$> n s))
+
+buildCofree
```

Comment:
It doesn't look like this is used, and it looks like a duplicate of `unfoldCofree` essentially. Is the idea just to save an extra allocation?

I would like to have only one, and deprecate the `unfoldCofree` if we decide to keep this.

Reply:
The problem with `unfoldCofree` is that the label and the functor are calculated independently, so they cannot share work. `buildCofree` supersedes `unfoldCofree`.

Reply:
Yeah, I'm happy to go with `buildCofree`, although it really is the natural unfold, of course, so I would prefer if we could use the name `unfoldCofree`. I suggest just adding a note that `unfoldCofree` will be replaced by `buildCofree` in a future release, please.